### PR TITLE
Add actions dropdown to object detail modal

### LINKED ISSUE #32321: Add actions dropdown to object detail modal
Epic: #32320
Product doc: [link](https://www.notion.so/Allow-basic-deletes-and-updates-on-object-detail-views-f05b298c3aba4a6ca883b2738d86824d)
Blocks: #32322 & #32323

----

- Model Detail views will get a new dropdown menu for “Actions” which will show options to update or delete that record using basic actions.

Conditions to show the dropdown:
- model must have actions enabled
- AND Update or Delete basic actions (or both) must exist and have not been archived

### PR DESCRIPTION
**Pull request by kamilmielnik**
Closes #32321

### Description

This is a part of an epic #32320 which has a dedicated branch: `32320-object-detail-actions`.

More comprehensive test suite will be added later: #32350.
If you have suggestions for additional tests please add them to #32350.

I recommend to [hide whitespace](https://github.com/metabase/metabase/assets/6830683/9c8c2f27-3cab-40c1-a753-c61608ed7449) when reviewing this PR.

### Design notes

- the button to open the actions menu is larger than the buttons for previous/next object
- in modals, the close button is only shown on viewports narrower than `960px`
- extra data (list of actions and databases) needs to be fetched when modal is opened for the first time for a given model: there is no loading or error state for this (just like for other loadable data in that modal) - the actions button will appear once actions are fetched

### Demo

https://github.com/metabase/metabase/assets/6830683/696ba955-caaf-4433-8bd2-2ee8efe49f4e

### CODE REVIEW

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx
Comment (kdoh) [inline frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx:451]: I think in general if we can it'd be ideal to use an value from the scale like `2` or something here rather than `1rem` so there's an association with the theme. Let me know if I'm missing some context though.

Comment (kamilmielnik) [reply frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx]: @kdoh Did you mean something like `<Flex align="center" p="md">`?
If not, can you please provide a code snippet?

Comment (kdoh) [reply frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx]: Yup that's the idea, just using a value from the spacing scale.

File: frontend/src/metabase/models/components/ModelDetailPage/ModelActionDetails/ModelActionDetails.tsx
Comment (kamilmielnik) [inline frontend/src/metabase/models/components/ModelDetailPage/ModelActionDetails/ModelActionDetails.tsx:42]: It was unused.

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx
Comment (kamilmielnik) [reply frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx]: Done: 3d58d697b8b1d2b5ac0f78f5889d28dbc57add45

Previously the horizontal spacing between these buttons was `0.5rem`.
The `gap` prop of `Flex` gives:
- `0.625rem` for `xs`
- `0.75rem` for `sm`
- `1rem` for `m`

I went with `xs` as it's the closest to `0.5rem`.

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.unit.spec.tsx
Comment (kamilmielnik) [inline frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.unit.spec.tsx:24]: These changes are needed because of `useActionListQuery` and `useDatabaseListQuery` usage in `ObjectDetail`.

I didn't want to use deprecated HOCs and I didn't want to introduce another wrapper component for `ObjectDetail` - this is the consequence.

Please let me know if something else is preferable.

File: frontend/src/metabase/common/hooks/use-action-list-query/use-action-list-query.ts
Comment (kamilmielnik) [inline frontend/src/metabase/common/hooks/use-action-list-query/use-action-list-query.ts:11]: I don't like the `Action` vs `WritebackAction` inconsistency in the naming, but:
- it's `entities/actions`, not `entities/writeback-actions`
- renaming `WritebackAction` is a material for another day

**Comment by kamilmielnik** [comment]
@kdoh I requested review from you mainly to have a look at "design notes" section in the PR description. In the implementation I went with how it works now for other data, but we might want to revisit these in the future.

File: frontend/src/metabase/common/hooks/use-action-list-query/use-action-list-query.ts
Comment (ranquild) [reply frontend/src/metabase/common/hooks/use-action-list-query/use-action-list-query.ts:11]: Agree, `WritebackAction` is used everywhere atm

File: frontend/src/metabase/visualizations/components/ObjectDetail/utils.unit.spec.ts
Comment (ranquild) [inline frontend/src/metabase/visualizations/components/ObjectDetail/utils.unit.spec.ts:35]: Creating metabase-lib entities directly is forbidden. The correct way is to use `createMockMetadata` and get them via `metabase.database(...)`. Unfortunately there is nothing preventing people from calling the constructor.

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.unit.spec.tsx
Comment (ranquild) [inline frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.unit.spec.tsx:23]: Could you also add basic tests that new menu items are shown when expected? Given that edge cases are tested within the utility, you'd likely need to have 2 tests here - 1 when no new menu items are shown and another one with all menu items

**Review by ranquild — APPROVED** [review]
Looks good. Left some comments about the tests. I believe this is something that can be fixed without another review, so approving now

**Review by kdoh — APPROVED** [review]
Looking solid so far! I read through the list of design notes you left and I think what we ought to do is add those to the main epic and reassess those as polish items once we have a round trip working. Think it's premature to focus on those just yet.

File: frontend/src/metabase/visualizations/components/ObjectDetail/utils.unit.spec.ts
Comment (kamilmielnik) [reply frontend/src/metabase/visualizations/components/ObjectDetail/utils.unit.spec.ts]: b0b7a3ac68d9d5752defe5d815a3f3e1d7f3930b

File: frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx
Comment (ranquild) [reply frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx]: I think these values were never approved, they are just mantine defaults, I don't think we could/should use them right now

Comment (kamilmielnik) [reply frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetail.tsx]: > I think these values were never approved, they are just mantine defaults, I don't think we could/should use them right now

Thanks! Updated in another branch: 28d03593aca9b1db9cd2fb4124f7d54b6e7f56af

**Pull request merged**
